### PR TITLE
Ollama integration

### PR DESCRIPTION
**Pull request by galshubeli**
Add Ollama to the list of available models.

## Summary by CodeRabbit

- **New Features**
	- Enhanced configuration handling for the Ollama generative model and chat session.
	- Added ability to retrieve chat history in the chat session.
	- Expanded Knowledge Graph with new "Actor" and "Movie" entities and updated test logic.
	- Improved testing workflow with health checks and logging for the Ollama service.

- **Bug Fixes**
	- Streamlined methods for better performance and reliability in the chat session and generative model.

- **Documentation**
	- Added docstrings for improved clarity on methods within the Ollama classes.

### CODE REVIEW

File: graphrag_sdk/models/ollama.py [hunk 1]
```diff
@@ -49,15 +94,31 @@ def ask(self, message: str) -> GenerationResponse:
...
+        Parse the model's response and extract content for the user.
 
-    def parse_generate_content_response(self, response: any) -> GenerationResponse:
+        Args:
+            response (any): The raw response from the model.
+
+        Returns:
+            GenerationResponse: Parsed response containing the generated text.
+        """
         print(response)
```
Comment (swilly22) [inline graphrag_sdk/models/ollama.py:109]: I don't think we should have this print.

**Review by swilly22 — CHANGES_REQUESTED** [review]

File: graphrag_sdk/models/ollama.py [hunk 2]
```diff
@@ -1,41 +1,86 @@
...
+        Returns:
+            GenerativeModel: The updated model instance.
+        """
         self.system_instruction = system_instruction
-        self.client = None
-        self._get_model()
+        self._connect_to_model()
         self.client.pull(self.model_name)
 
         return self
```
[the remark this answers is not in the document]
Comment (galshubeli) [reply graphrag_sdk/models/ollama.py:60]: Yes, you're right that it's not the correct way, but I'm changing it in a separate PR for all models.

File: graphrag_sdk/models/ollama.py [hunk 3]
```diff
@@ -66,6 +127,15 @@ def to_json(self) -> dict:
...
+        Deserialize a JSON object to create an instance of OllamaGenerativeModel.
+
+        Args:
+            json (dict): The serialized JSON data.
+
+        Returns:
+            GenerativeModel: A new instance of the model.
+        """
         return OllamaGenerativeModel(
             model_name=json["model_name"],
```
[the remark this answers is not in the document]
Comment (galshubeli) [reply graphrag_sdk/models/ollama.py:139]: I think it should be included in the 'cleanup' PR. Let me know if you believe it should be here instead.

File: graphrag_sdk/models/ollama.py [hunk 4]
```diff
@@ -1,41 +1,86 @@
...
 
     def __init__(
         self,
         model_name: str,
-        generation_config: GenerativeModelConfig | None = None,
-        system_instruction: str | None = None,
-        host: str = None,
+        generation_config: Optional[GenerativeModelConfig] = None,
+        system_instruction: Optional[str] = None,
+        host: Optional[str] = None,
```
Comment (galshubeli) [inline graphrag_sdk/models/ollama.py:24]: host, port?

File: graphrag_sdk/models/ollama.py [hunk 5]
```diff
@@ -75,45 +144,70 @@ def from_json(json: dict) -> "GenerativeModel":
...
 class OllamaChatSession(GenerativeModelChatSession):
-
-    _history = []
-
-    def __init__(self, model: OllamaGenerativeModel, args: dict | None = None):
+    """
+    A chat session for interacting with the Ollama model, maintaining conversation history.
+    """
+    
+    def __init__(self, model: OllamaGenerativeModel, args: Optional[dict] = None):
```
Comment (galshubeli) [inline graphrag_sdk/models/ollama.py:151]: remove

File: graphrag_sdk/models/ollama.py [hunk 6]
```diff
@@ -75,45 +144,70 @@ def from_json(json: dict) -> "GenerativeModel":
...
-        content = self._model.parse_generate_content_response(response)
-        self._history.append({"role": "user", "content": message})
-        self._history.append({"role": "assistant", "content": content.text})
+        content = self._model._parse_generate_content_response(response)
+        self._chat_history.append({"role": "assistant", "content": content.text})
         return content
+    
+    def _get_generation_config(self, output_method: OutputMethod):
+        """
+        Adjust the generation configuration based on the output method.
```
Comment (galshubeli) [inline graphrag_sdk/models/ollama.py:200]: name

**Review by swilly22 — APPROVED** [review]

**Pull request merged**